infra/src/main/kotlin/fi/fta/geoviite/infra/publication/PublicationService.kt

```diff
@@ -349,75 +349,39 @@ class PublicationService @Autowired constructor(
     }
 
     @Transactional(readOnly = true)
-    fun getRevertRequestDependencies(publishRequestIds: PublishRequestIds): PublishRequestIds {
-        logger.serviceCall("getRevertRequestDependencies", "publishRequestIds" to publishRequestIds)
+    fun getRevertRequestDependencies(requestIds: PublishRequestIds): PublishRequestIds {
+        logger.serviceCall("getRevertRequestDependencies", "publishRequestIds" to requestIds)
 
-        val draftOnlyTrackNumbers = (publishRequestIds.referenceLines.mapNotNull { id ->
-            referenceLineService.get(DRAFT, id)?.trackNumberId?.let { trackNumberId ->
-                if (!trackNumberService.officialExists(trackNumberId)) trackNumberId else null
-            }
-        } + publishRequestIds.trackNumbers.filterNot { id -> trackNumberDao.officialExists(id) }).toSet()
-        val locationTracks = publishRequestIds.locationTracks.toSet() + draftOnlyTrackNumbers.flatMap { trackNumberId ->
-            locationTrackDao.fetchOnlyDraftVersions(false, trackNumberId)
-        }.map { lt -> lt.id }
-        val kmPosts = publishRequestIds.kmPosts.toSet() + draftOnlyTrackNumbers.flatMap { trackNumberId ->
-            kmPostDao.fetchOnlyDraftVersions(false, trackNumberId)
+        val referenceLineTrackNumberIds = referenceLineService.getMany(DRAFT, requestIds.referenceLines).map { rlId ->
+            rlId.trackNumberId
+        }
+        val revertTrackNumberIds = mutableSetOf<IntId<TrackLayoutTrackNumber>>()
+        val draftOnlyTrackNumberIds = mutableSetOf<IntId<TrackLayoutTrackNumber>>()
+        trackNumberService.getMany(DRAFT, referenceLineTrackNumberIds + requestIds.trackNumbers).forEach { tnId ->
+            if (tnId.getDraftType() != DraftType.OFFICIAL) revertTrackNumberIds.add(tnId.id as IntId)
+            if (tnId.getDraftType() == DraftType.NEW_DRAFT) draftOnlyTrackNumberIds.add(tnId.id as IntId)
+        }
```

Comment:
Nitpick, mutta näiden listojen muodostuksen saisi hoidettua myös niin, että tekisi tuon `getMany`:n erilliseen apumuuttujaan ja sen jälkeen filtteröisi sen tuloksen kahdella eri filtterillä (plus loppuun `.distinct()` jos setin tuoma uniikkius on tässä oleellista.) Enemmän looppausta, mutta toisaalta tällöin ei tarvisi mutableSettejä.

Reply:
Njoo, koitin jostain syystä välttää tuplaiterointia, mutta se on kyllä ihan sama iteroida kahdesti tai tarkastaa kaksi asiaa yhdessä iteroinnissa. Voisin vaihtaa tämän siihen malliin, on se nätimpi.